src/Umbraco.Infrastructure/Security/MembersUserStore.cs

```diff
@@ -0,0 +1,703 @@
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Identity;
+using Umbraco.Core;
+using Umbraco.Core.Mapping;
+using Umbraco.Core.Models;
+using Umbraco.Core.Models.Identity;
+using Umbraco.Core.Scoping;
+using Umbraco.Core.Services;
+
+namespace Umbraco.Infrastructure.Security
+{
+    /// <summary>
+    /// A custom user store that uses Umbraco member data
+    /// </summary>
+    public class MembersUserStore : UserStoreBase<MembersIdentityUser, IdentityRole<string>, string, IdentityUserClaim<string>, IdentityUserRole<string>, IdentityUserLogin<string>, IdentityUserToken<string>, IdentityRoleClaim<string>>
+    {
+        private readonly IMemberService _memberService;
+        private readonly UmbracoMapper _mapper;
+        private readonly IScopeProvider _scopeProvider;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MembersUserStore"/> class for the members identity store
+        /// </summary>
+        /// <param name="memberService">The member service</param>
+        /// <param name="mapper">The mapper for properties</param>
+        /// <param name="scopeProvider">The scope provider</param>
+        /// <param name="describer">The error describer</param>
+        /// 
+        public MembersUserStore(IMemberService memberService, UmbracoMapper mapper, IScopeProvider scopeProvider, IdentityErrorDescriber describer)
+        : base(describer)
+        {
+            _memberService = memberService ?? throw new ArgumentNullException(nameof(memberService));
+            _mapper = mapper;
+            _scopeProvider = scopeProvider;
+        }
+
+        /// <summary>
+        /// Not supported in Umbraco
+        /// </summary>
+        /// <inheritdoc />
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public override IQueryable<MembersIdentityUser> Users => throw new NotImplementedException();
+
+        /// <inheritdoc />
+        public override Task<string> GetNormalizedUserNameAsync(MembersIdentityUser user, CancellationToken cancellationToken) => GetUserNameAsync(user, cancellationToken);
+
+        /// <inheritdoc />
+        public override Task SetNormalizedUserNameAsync(MembersIdentityUser user, string normalizedName, CancellationToken cancellationToken) => SetUserNameAsync(user, normalizedName, cancellationToken);
+
+        /// <inheritdoc />
+        public override Task<IdentityResult> CreateAsync(MembersIdentityUser user, CancellationToken cancellationToken)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+            ThrowIfDisposed();
+            if (user == null)
+            {
+                throw new ArgumentNullException(nameof(user));
+            }
+
+            // create member
+            // TODO: are we keeping this method? The user service creates the member directly
+            // but this way we get the member type by alias first
+            IMember memberEntity = _memberService.CreateMember(
```

Comment:
Yes I agree, I felt like I wanted to replace it but was trying to stick to 1-to-1 changes. So this would be a great future task to replace these methods.

Reply:
cc @bergmania can we create a new task for this pls?

Reply:
I have one internal task for that